### PR TITLE
feat: Make security tokens optional

### PR DESCRIPTION
**Pull request by wbreza**
Security tokens are a great way to protect sensitive data from being stored in plain text within your project file, but at the same time can be more difficult to use when sharing projects with team members.

In this feature security tokens are still enabled by default but give the project author the ability to turn off security tokens which then stores all provider options in plain text.

Fixes: #835 #security

image

### CODE REVIEW

File: src/common/mockFactory.ts [hunk 1]
```diff
@@ -277,6 +277,7 @@ export default class MockFactory {
             id: `project-${name}`,
             name: `Project ${name}`,
             version: appInfo.version,
+            useSecurityToken: true,
```
Comment (tessmichi) [inline src/common/mockFactory.ts:280]: is there no mock for not using security token? if not, how is that tested?

**Review by tessmichi — APPROVED** [review]
lgtm

File: src/common/mockFactory.ts (hunk 1 above)
Comment (wbreza) [reply src/common/mockFactory.ts:280]: No, but i have a test that modifies the value after getting it from the mock.

**Review by tbarlow12 — APPROVED** [review]
ship it

**Review by jding05 — APPROVED** [review]

**Pull request merged**